dd-sdk-android/src/main/kotlin/com/datadog/android/webview/internal/log/WebViewLogEventConsumer.kt

```diff
@@ -36,22 +37,23 @@ internal class WebLogEventConsumer(
         }
     }
 
-    private fun map(event: JsonObject): WebLogEvent? {
-        return try {
+    private fun map(event: JsonObject): WebViewLogEvent? {
+        try {
             addDdTags(event)
-            val logEvent = WebLogEvent.fromJson(event.toString())
+            val logEvent = WebViewLogEvent.fromJson(event.toString())
             val rumContext = rumContextProvider.getRumContext()
-            if (rumContext != null) {
+            val correctedDate = dateCorrector.correctDate(logEvent.date) ?: return null
+            return if (rumContext != null) {
                 val resolvedProperties = logEvent.additionalProperties.toMutableMap()
                 resolvedProperties[LogAttributes.RUM_APPLICATION_ID] = rumContext.applicationId
                 resolvedProperties[LogAttributes.RUM_SESSION_ID] = rumContext.sessionId
-                logEvent.copy(additionalProperties = resolvedProperties)
+                logEvent.copy(additionalProperties = resolvedProperties, date = correctedDate)
```

Comment:
should(n't) we keep `offsets` cache as we do in Web RUM Events so that we correct the dates per RUM View?
(we can get active RUM View from `rumContext`)

Reply:
I am not sure this is needed here as log events are singular, they do not require that. At least we are not doing this on our end with the native log events.